galley/pkg/crd/validation/config.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"reflect"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/api/admissionregistration/v1beta1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	admissionregistration "k8s.io/client-go/kubernetes/typed/admissionregistration/v1beta1"
+
+	"istio.io/istio/pkg/log"
+)
+
+func (wh *Webhook) reconcileWebhookConfiguration() {
+	client := wh.clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations()
+	updated, err := reconcileWebhookConfigurationHelper(client, wh.webhookConfiguration)
+	if err != nil {
+		log.Errorf("%v validatingwebhookconfiguration update failed: %v",
+			wh.webhookConfiguration.Name, err)
+		reportValidationConfigUpdateError(err)
+	} else if updated {
+		log.Infof("%v validatingwebhookconfiguration updated", wh.webhookConfiguration.Name)
+		reportValidationConfigUpdate()
+	}
+}
+
+func reconcileWebhookConfigurationHelper(
```

Comment:
nit: more descriptive names can help readability here:
upsertWebhookConfig, patchOrCreateWebhookConfig etc.

Also comments that explain the high-level logic would be quite useful.